@here/harp-mapview/test/CopyrightProvidersTest.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// tslint:disable:only-arrow-functions
+// Mocha discourages using arrow functions, see https://mochajs.org/#arrow-functions
+
+import { expect } from "chai";
+import * as sinon from "sinon";
+
+import { GeoBox, GeoCoordinates } from "@here/harp-geoutils";
+import { TransferManager } from "@here/harp-transfer-manager";
+import { CopyrightInfo } from "../lib/copyrights/CopyrightInfo";
+import { UrlCopyrightProvider } from "../lib/copyrights/UrlCopyrightProvider";
+
+describe("CopyrightProviders", function() {
+    describe("#UrlCopyrightProvider", function() {
+        async function getCopyrights(geoBox: GeoBox, level: number): Promise<CopyrightInfo[]> {
+            const provider = new UrlCopyrightProvider("", "normal");
+            return provider.getCopyrights(geoBox, level);
+        }
+
+        describe("#init", function() {
+            it("Should return default copyrights if failed to load data", async function() {
+                const fakeJson = sinon.fake.rejects(new Error("error"));
+
+                sinon.replace(TransferManager.prototype, "downloadJson", fakeJson);
+                const copyrights = await getCopyrights(
+                    new GeoBox(new GeoCoordinates(2, 2), new GeoCoordinates(3, 3)),
+                    10
+                );
+                sinon.restore();
+
+                expect(copyrights).to.deep.equal([]);
+            });
+
+            it("Should get copyrights more than once", async function() {
```

Comment:
Asserts are missing is missing in test and description id inaccuare:
`Shouldn't ...`  ? - i would expect this that we test that, `UrlCopyrightProvider` requests CI only once
Or .. `Should` ... in case of failure

Reply:
Updated this test's name and added assertion. Basically, purpose of this test is to improve code coverage (to get into branches with cached promises).